crates/geometry/src/line.rs

```diff
@@ -90,21 +77,35 @@ impl<Frame, const DIMENSION: usize> Distance<Point<Frame, DIMENSION>> for Line<F
     }
 }
 
+impl<Frame, const DIMENSION: usize> Default for Line<Frame, DIMENSION> {
+    fn default() -> Self {
+        Self {
+            point: Default::default(),
+            direction: Vector::zeros(),
+        }
+    }
+}
+
 impl<From, To, const DIMENSION: usize, Inner> Mul<Line<From, DIMENSION>>
     for Transform<From, To, Inner>
 where
-    Self: Mul<Point<From, DIMENSION>, Output = Point<To, DIMENSION>> + Copy,
+    Self: Mul<Point<From, DIMENSION>, Output = Point<To, DIMENSION>>
+        + Mul<Vector<From, DIMENSION>, Output = Vector<To, DIMENSION>>
+        + Copy,
 {
     type Output = Line<To, DIMENSION>;
 
-    fn mul(self, right: Line<From, DIMENSION>) -> Self::Output {
-        Line(self * right.0, self * right.1)
+    fn mul(self, line: Line<From, DIMENSION>) -> Self::Output {
+        Line {
+            point: self * line.point,
+            direction: self * line.direction,
+        }
     }
 }
 
 impl<Frame, const DIMENSION: usize> PartialEq for Line<Frame, DIMENSION> {
     fn eq(&self, other: &Self) -> bool {
-        (self.0 == other.0 && self.1 == other.1) || (self.0 == other.1 && self.1 == other.0)
+        self.point == other.point && self.direction == other.direction
```

Comment:
The previous implementation considered a line with negative `direction` to be the same. Is it correct to drop this assumption? Also, it is possible to define the same line with multiple `point`s and `direction`s like scaling them, or moving the point along the direction.

Reply:
IMO two lines are the same if their normalized direction is the same (also with the same sign, since the behavior of `signed_distance_to` is inconsistent otherwise) and if the point of one line lies also on the other one. I am not sure whether this may result in problems with floating point inaccuracies, i.e. whether `PartialEq` should be implemented at all for `Line`. What is your opinion on this?

Reply:
Is it used somewhere? And when yes, for what?

Reply:
No, it is not used anywhere. Though `RelativeEq` is used in some test cases, so maybe we should drop the `PartialEq` implementation and do `RelativeEq` according to the prior definition.

Reply:
I now updated the implementations to consider lines as (relative) equal if their normalized directions are the same and the point of one line also lies on the other one.